hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -43,6 +43,11 @@ public HiveCatalog(Configuration conf) {
     this.conf = conf;
   }
 
+  @Override
+  protected String name() {
+    return conf.get("hive.metastore.uris");
```

Comment:
I agree with @rdsr , using metastore uri seems a bit strange, maybe we could use conf to specify one. For example if it adapts to Spark DSv2 catalog interface, it could leverage Spark to pass in a catalog name.

Reply:
What about defaulting this for HMS to something like "hive"?

Reply:
I think that would also be OK.

Reply:
@jerryshao, what about `HadoopCatalog`? `hdfs`?

Reply:
Maybe just `hadoop`, I guess hdfs may not be the only option in Hadoop area.